Sources/Apollo/GraphQLResult.swift

```diff
@@ -3,6 +3,28 @@ import Foundation
 import ApolloAPI
 #endif
 
+/// Metadata about the a returned result.
+ public struct GraphQLResultMetadata {
+   /// The oldest date that data in the result was updated.
+   public let maxAge: Date
+   public let id = UUID()
+
+   init(maxAge: Date = Date()) {
+     self.maxAge = maxAge
+   }
+ }
+
+extension GraphQLResultMetadata: Hashable {
+
+  public func hash(into hasher: inout Hasher) {
+    hasher.combine(self.id)
+  }
+
+  public static func == (lhs: GraphQLResultMetadata, rhs: GraphQLResultMetadata) -> Bool {
+   return lhs.maxAge == rhs.maxAge
+  }
```

Comment:
❗ This violates the semantics of hashable & equatable.

You should be using the same properties in both.

Reply:
done